src/vm/node_modules/VM.js

```diff
@@ -518,6 +518,38 @@ function getZpools(log, callback)
     });
 }
 
+function getOptimalVolblocksize(log, callback)
+{
+    var args = ['-C'];
+    var cmd = '/usr/sbin/zdb';
+
+    assert(log, 'no logger passed to zpoolRequiresLargerVolblocksize()');
+
+    traceExecFile(cmd, args, log, 'zdb-c',
+        function (error, stdout, stderr) {
+
+        if (error) {
+            log.error('Unable to get zdb pool data');
+            callback(error, {'stdout': stdout, 'stderr': stderr});
+            return;
+        }
+
+        var hasAshift9  = stdout.match('ashift: 9');
+        var hasAshift12 = stdout.match('ashift: 12');
+        var hasRaidZ = stdout.match("type: 'raidz'")
+
+        // This is an approximation: we only increase the blocksize if we know
+        // that the entire pool(s) is using 4K pages. If even one disk is using
+        // 512b pages (e.g. the data disks are 4K, but the slog is 512b) then
+        // we stick with the default.
+        var volblocksize = hasAshift12 && !hasAshift9 && hasRaidZ ?
+            16384 :
```

Comment:
You should set it to 32k, per your writeup.

Reply:
We haven't run with 32k in production here. If you believe switching to 32k here is reasonable though, I'm fine with that.

Reply:
Your numbers suggest 32k will spike well in space-efficiency in places that are advice-giving worth, and not harm the others relative to 16k.  Why not?